service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +46,133 @@ public MapRequester(Vertx vertx) {
-                                            + updateTo.name());
-                                    this.tokenFuture = updateToken();
-                                    updateLayers(streetFids, updateTo, this.tokenFuture)
-                                        .onSuccess((V) -> promise.complete())
-                                        .onFailure(promise::fail);
-                                  } else if (responseBody.containsKey("updateResults")) {
-                                    logger.info(
-                                        "Successfully updated blocks to " + updateTo.name());
-                                    // Check successes
-                                    promise.complete();
-                                  } else {
-                                    // throw error
-                                    logger.error(
-                                        "Update street request responded with unrecognized response body: "
-                                            + responseBody);
+                                  if (responseBody.containsKey("features")) {
+                                    List<Map> features =
+                                        responseBody.getJsonArray("features").getList();
+                                    JsonArray updateJson = new JsonArray();
+                                    features.forEach(
+                                        feature -> {
+                                          Map attributes = (Map) feature.get("attributes");
+                                          String FID = attributes.get("FID").toString();
+                                          updateJson.add(
+                                              JsonObject.mapFrom(new MapRequest(FID, updateTo)));
+                                        });
+                                    promise.complete(updateJson);
+                                    return;
                                   }
-                                } else {
-                                  logger.error(
-                                      "Update street response request responded with non-200 status code: "
-                                          + httpResponse.statusCode());
                                 }
                               } else {
-                                logger
-                                    .atError()
-                                    .withThrowable(ar.cause())
-                                    .log("Error sending update street request", ar.cause());
-                                promise.fail(ar.cause());
+                                promise.fail("Error fetching Block FIDs");
                               }
-                            });
+                            }));
                   });
             });
 
-    return updateFuture;
+    Future<Void> updateMapFuture =
```

Comment:
the diff is showing it weird, but this function actually isn't new, it's the existing one. The new function is above, getBlockFIDsFuture. But review this too cause it changed slightly.